plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/totals/coupon/index.tsx

```diff
@@ -53,7 +52,7 @@ export const TotalsCoupon = ( {
 	const { validationErrorId } = useSelect( ( select ) => {
 		const store = select( VALIDATION_STORE_KEY );
 		return {
-			validationErrorId: store.getValidationErrorId( textInputId ),
+			validationErrorId: store.getValidationErrorId( instanceId ),
```

Comment:
With the old implementation, the value of the `textInputId` const was `wc-block-components-totals-coupon__input-0`. Then, this string was passed to the `store.getValidationErrorId` function to return the `validationErrorId` value.

However, looking at the [implementation of the `getValidationErrorId function`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/data/validation/selectors.ts#L24), we can see that it returns `undefined` if the `state` object doesn't have a property with the `wc-block-components-totals-coupon__input-0` name.

This also explains why the `validationErrorId` [returns `undefined`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/packages/components/validation-input-error/index.tsx#L28) from within the `ValidationInputError` component. Since it's `undefined`, the `id` attribute is empty and there is no way to [associate the error message with the coupon input](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/totals/coupon/index.tsx#L104).

To fix it, the `"coupon"` string should be passed to the `store.getValidationErrorId` instead of `wc-block-components-totals-coupon__input-0` since this is the property name set to the [`state` object](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/base/context/hooks/cart/use-store-cart-coupons.ts#L94). In order to pass it, the `TotalsCoupon` component should no longer be wrapped by the `withInstanceId` HOC, so the `instanceId` prop can be manually passed.